src/components/network/network.js

```diff
@@ -607,15 +607,18 @@ export default class Network {
                 break;
             case 'VOLTAGE_LEVEL':
                 const substationId =
-                    this.voltageLevelsById.get(equipmentId).substationId;
-                let voltageLevelsOfSubstation =
-                    this.substationsById.get(substationId).voltageLevels;
-                voltageLevelsOfSubstation = voltageLevelsOfSubstation.filter(
-                    (l) => l.id !== equipmentId
-                );
-
-                this.substationsById.get(substationId).voltageLevels =
-                    voltageLevelsOfSubstation;
+                    this.voltageLevelsById.get(equipmentId)?.substationId;
```

Comment:
It seems substationId is only needed inside the if.
I would prefer defining `const voltageLevel = this.voltageLevelsById.get(equipmentId)`
and testing its voidness as the condition of the the block.
The same with the substation : `const substation = this.substationsById.get(substationId)`